scipy/sparse/linalg/_eigen/tests/test_svds.py

```diff
@@ -424,23 +420,20 @@ def test_svd_rng_2(self, rng):
             assert_allclose(res1a[idx], res2a[idx], rtol=1e-15, atol=2e-16)
         _check_svds(A, k, *res1a)
 
-    @pytest.mark.parametrize("rng", (None,
-                                     np.random.default_rng(0)))
     @pytest.mark.filterwarnings("ignore:Exited",
                                 reason="Ignore LOBPCG early exit.")
-    def test_svd_rng_3(self, rng):
+    def test_svd_rng_3(self):
         n = 100
         k = 5
 
-        tmp_rng = np.random.default_rng(0)
-        A = tmp_rng.random((n, n))
-
-        rng = copy.deepcopy(rng)
+        rng1 = np.random.default_rng(0)
+        rng2 = np.random.default_rng(234832)
+        A = rng1.random((n, n))
```

Comment:
It looks like we don't need different `rng`s here. One would do because the state will change after the first call, right?

Reply:
Sure, but it also can't hurt, this is slightly more explicit. 

This whole test could also be merged with the `test_svd_rng_2` above it. I limited myself to cleaning up the most obvious problems and fixing the test failure.